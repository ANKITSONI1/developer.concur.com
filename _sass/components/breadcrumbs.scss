// bootstrap overrides
$breadcrumb-bg: none;
$breadcrumb-padding-vertical: 10px;
$breadcrumb-padding-horizontal: 0;
$breadcrumb-color: #555;

.breadcrumbs {
  background: #fafafa;
  border-bottom: solid 1px #eee;

  h1 {
    margin-top: 16px;
    font-size: 22px;
  }

  .breadcrumb {
    position: relative;
    top: 10px;
  }
}


@media (max-width: $grid-float-breakpoint-max) {
  .breadcrumbs h1,
  .breadcrumbs .breadcrumb {
    width: 100%;
    text-align: center;
  }
}
