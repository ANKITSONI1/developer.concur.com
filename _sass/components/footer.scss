.footer {
  padding: 30px 0;
  background-color: $off-black;
  color: #eee;

  p {
    color: #eee;
  }
}

.headline {
  margin: 10px 0 25px 0;
  border-bottom: 2px solid $concur-blue;

  h2, h3, h4 {
    padding-bottom: 5px;
    color: #eee;
    border-bottom: 2px solid $concur-blue;
  }
}

.footer-copyright {
  margin-top: 20px;
}

.social-icons {
  float: right;
}

@media (max-width: $grid-float-breakpoint-max) {
  .footer-copyright,
  .social-icons {
    float: none;
    text-align: center;
  }
}

.footer-github {
  margin: 30px 0;

  span {
    margin: 0 10px;
  }

  * {
    color: #eee;
  }
}

.company-description {
  margin-bottom: 65px;
}

.link-list li {
  border-top: solid 1px #353535;

  &:first-child {
    border-top: none !important;
  }

  i {
    color: #bbb;
    float: right;
    margin-top: 10px;
  }
}

.link-list a {
  color: #eee;
  font-size: 11px;
  padding: 6px 0px;
  display: inline-block;
  text-transform: uppercase;

  &:hover {
    color: #eee;
  }
}
