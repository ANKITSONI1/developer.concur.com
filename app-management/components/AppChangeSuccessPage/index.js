import { connect } from 'react-redux';
import AppChangeSuccessPage from './AppChangeSuccessPage';

export function mapStateToProps(state) {
  return state.appChange;
}

export default connect(
  mapStateToProps,
  null,
)(AppChangeSuccessPage);
