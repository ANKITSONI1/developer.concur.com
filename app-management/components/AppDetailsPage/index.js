/* eslint-env browser */

import { connect } from 'react-redux';
import AppDetailsPage from './AppDetailsPage';
import { fetchAppDetails } from '../../actions/appDetails';
import { appChangeFailure, generateAppSecret, putApp } from '../../actions/appChange';
import { sharedHelpers } from '../../utils/actionHelpers';

export function mapStateToProps(state) {
  return {
    appChange: state.appChange,
    appDetails: state.appDetails,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    fetchAppDetails: id => dispatch(fetchAppDetails(id)),
    generateSecretHandler: (id, appName) => {
      const confirmPrompt = window.prompt('Type the name of your app to confirm clientSecret regeneration. This CANNOT be undone.');

      if (confirmPrompt === appName) {
        dispatch(generateAppSecret(id));
      } else {
        dispatch(appChangeFailure('Application names do not match.'));
      }
    },
    putAppHandler: (app, appName) => {
      const confirmPrompt = window.prompt('Type the original name of your app to confirm update. This will regenerate your clientSecret, which CANNOT be undone.');

      if (confirmPrompt === appName) {
        const composedApp = sharedHelpers.composeApp(app);
        dispatch(putApp(composedApp));
      } else {
        dispatch(appChangeFailure('Application names do not match.'));
      }
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AppDetailsPage);
