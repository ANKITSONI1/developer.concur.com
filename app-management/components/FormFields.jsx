/*
 * Wrapper components to render SkyForm fields with validation errors
 */

import React, { PropTypes } from 'react';
import Select from 'react-select';
import { Field } from 'redux-form';

export const FieldError = ({ error, ariaText }) => (
  <span id={ariaText} className="help-block">{error}</span>
);

FieldError.propTypes = {
  error: PropTypes.any,
  ariaText: PropTypes.string,
};

FieldError.defaultProps = {
  error: null,
  ariaText: '',
};

export const FieldHelp = ({ children }) => (
  <p>
    <small>
      <i className="fa fa-question-circle" />&nbsp;
      {children}
    </small>
  </p>
);

FieldHelp.propTypes = {
  children: PropTypes.any.isRequired,
};

export const InputField = ({ input, type, label, placeholder, meta: { touched, error } }) => {
  const { name } = input;
  const errorClass = touched && error ? 'has-error' : '';
  const ariaText = `${name}-help`;

  return (
    <div className={`form-group ${errorClass}`}>
      <label htmlFor={name} className="control-label">{label}</label>
      <input
        {...input}
        id={name}
        type={type}
        placeholder={placeholder}
        className="form-control"
        aria-describedby={ariaText}
      />
      {touched && error && <FieldError error={error} ariaText={ariaText} />}
    </div>
  );
};

InputField.propTypes = {
  input: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  meta: PropTypes.object.isRequired,
};

InputField.defaultProps = {
  placeholder: '',
};

export const TextareaField = ({ input, label, placeholder, meta: { touched, error } }) => {
  const { name } = input;
  const errorClass = touched && error ? 'has-error' : '';
  const ariaText = `${name}-help`;

  return (
    <div className={`form-group ${errorClass}`}>
      <label htmlFor={name} className="control-label">{label}</label>
      <textarea
        {...input}
        id={name}
        placeholder={placeholder}
        className="form-control"
        aria-describedby={ariaText}
      />
      {touched && error && <FieldError error={error} ariaText={ariaText} />}
    </div>
  );
};

TextareaField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  meta: PropTypes.object.isRequired,
};

TextareaField.defaultProps = {
  placeholder: '',
};

export const SelectField = ({ input, label, placeholder, options, meta: { touched, error } }) => {
  const { name } = input;
  const errorClass = touched && error ? 'has-error' : '';
  const ariaText = `${name}-help`;

  return (
    <div className={`form-group ${errorClass}`}>
      <label htmlFor={name}>{label}</label>
      <Select
        {...input}
        id={name}
        placeholder={placeholder}
        options={options}
        onBlur={() => input.onBlur(input.value)}
        aria-describedby={ariaText}
      />
      {touched && error && <FieldError error={error} ariaText={ariaText} />}
    </div>
  );
};

SelectField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  options: PropTypes.array.isRequired,
  meta: PropTypes.object.isRequired,
};

SelectField.defaultProps = {
  placeholder: '',
};

export const MultiselectField = ({
  input,
  label,
  placeholder,
  options,
  meta: { touched, error },
}) => {
  const { name } = input;
  const errorClass = touched && error ? 'has-error' : '';
  const ariaText = `${name}-help`;

  return (
    <div className={`form-group ${errorClass}`}>
      <label htmlFor={name}>{label}</label>
      <Select
        {...input}
        id={name}
        multi
        placeholder={placeholder}
        options={options}
        clearable={false}
        onBlur={() => input.onBlur(input.value)}
      />
      {touched && error && <FieldError error={error} ariaText={ariaText} />}
    </div>
  );
};

MultiselectField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  options: PropTypes.array.isRequired,
  meta: PropTypes.object.isRequired,
};

MultiselectField.defaultProps = {
  placeholder: '',
};

// Due to the checkbox being wrapped by a label, this component doesn't use FieldWrapper
export const CheckboxField = ({ input, meta: { touched, error }, children }) => {
  const { name } = input;
  const errorClass = touched && error ? 'has-error' : '';
  const ariaText = `${name}-help`;

  return (
    <div className={`form-group ${errorClass}`}>
      <label htmlFor={name}>
        <input {...input} id={name} type="checkbox" aria-describedby={ariaText} />
        <i />&nbsp;
        {children}
      </label>
      {touched && error && <FieldError error={error} ariaText={ariaText} />}
    </div>
  );
};

CheckboxField.propTypes = {
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  children: PropTypes.any.isRequired,
};

export const RedirectUriField = ({
  input,
  label,
  placeholder,
  canDelete,
  onClick,
  meta: { touched, error },
}) => {
  const { name } = input;
  const errorClass = touched && error ? 'has-error' : '';
  const ariaText = `${name}-help`;

  return (
    <div className={`form-group ${errorClass}`}>
      <label htmlFor={name} className="control-label">{label}</label>
      <div className={canDelete ? 'input-group' : ''}>
        <input
          {...input}
          id={name}
          type="url"
          placeholder={placeholder}
          className="form-control"
          aria-describedby={ariaText}
        />
        {canDelete ? (
          <div className="input-group-addon uri-delete" onClick={onClick} title="Delete">
            <i className="fa fa-times-circle" />
          </div>
        ) : null}
      </div>
      {touched && error && <FieldError error={error} ariaText={ariaText} />}
    </div>
  );
};

RedirectUriField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  canDelete: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  meta: PropTypes.object.isRequired,
};

RedirectUriField.defaultProps = {
  label: '',
  placeholder: '',
};

export const RedirectUris = ({ fields }) => (
  <div className="redirect-uris">
    {fields.map((uri, idx) => (
      <Field
        key={idx}
        component={RedirectUriField}
        name={uri}
        label={idx === 0 ? 'Redirect URI' : null}
        placeholder="https://example-uri.com"
        canDelete={fields.length > 1}
        onClick={() => fields.remove(idx)}
      />
    ))}
    <a
      className="btn small dark-green uri-add"
      onClick={() => fields.push('')}
    >
      Add URI
    </a>
  </div>
);

RedirectUris.propTypes = {
  fields: PropTypes.object.isRequired,
};
