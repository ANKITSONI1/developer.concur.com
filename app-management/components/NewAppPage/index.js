import { connect } from 'react-redux';
import NewAppPage from './NewAppPage';
import { postApp } from '../../actions/appChange';
import { sharedHelpers } from '../../utils/actionHelpers';

export function mapStateToProps(state) {
  return state.appChange;
}

export function mapDispatchToProps(dispatch) {
  return {
    handleSubmit: (app) => {
      const composedApp = sharedHelpers.composeApp(app);
      dispatch(postApp(composedApp));
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NewAppPage);
