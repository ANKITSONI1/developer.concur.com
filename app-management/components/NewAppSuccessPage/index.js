import { connect } from 'react-redux';
import NewAppSuccessPage from './NewAppSuccessPage';

export function mapStateToProps(state) {
  return state.newApp;
}

export default connect(
  mapStateToProps,
  null,
)(NewAppSuccessPage);
