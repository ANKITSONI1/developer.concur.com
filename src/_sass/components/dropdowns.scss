.dropdown-menu {
  min-width: 200px;
  padding: 0;

  li {
    min-width: 250px;
  }

  li > a {
    font-weight: 300;
    padding: 5px 15px;

    &:hover,
    &:focus {
      color: $white;
      background-color: $dark-gray;
    }
  }

  &.two-column-menu {
    min-width: 450px;

    .col-sm-6:nth-child(1) {
      padding-right: 0 !important;
    }

    .col-sm-6:nth-child(2) {
      padding-left: 0 !important;
    }

    li {
      min-width: 200px;
    }
  }

  .multi-column-dropdown {
    list-style-type: none;
    margin: 0px;
    padding: 0px;

    li > a {
      display: block;
      line-height: 1.428571429;
      color: #333333;

      &:hover {
        background-color: #999;
        color: white;
        text-decoration: none;
      }
    }
  }
}

@media (min-width: $grid-float-breakpoint) {
  ul.nav li.dropdown:hover > ul.dropdown-menu {
    display: block;
  }
}

@media (max-width: $grid-float-breakpoint-max) {
  .dropdown-menu > li > a {
    border-bottom: 1px solid $off-black;
  }
}
