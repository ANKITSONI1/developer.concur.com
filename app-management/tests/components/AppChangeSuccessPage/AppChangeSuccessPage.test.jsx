import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';
import { shallowToJson } from 'enzyme-to-json';

import AppChangeSuccessPage from '../../../components/AppChangeSuccessPage/AppChangeSuccessPage';
import { mapStateToProps } from '../../../components/AppChangeSuccessPage';
import appReducer from '../../../reducers';
import appFactory from '../../app.mock';

const store = createStore(appReducer);

describe('<AppChangeSuccessPage />', () => {
  it('should render a not found page if no app or clientSecret are provided', () => {
    const app = {};
    const clientSecret = '';
    const page = shallow(<AppChangeSuccessPage app={app} clientSecret={clientSecret} />);

    expect(shallowToJson(page)).toMatchSnapshot();
  });

  it('should render the app clientId, clientSecret, and name if provided', () => {
    const app = appFactory('test-id');
    const clientSecret = 'a-client-secret';
    const page = shallow(<AppChangeSuccessPage app={app} clientSecret={clientSecret} />);

    expect(shallowToJson(page)).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('returns entire appChange state', () => {
      const state = mapStateToProps(store.getState());

      expect(state).toEqual(store.getState().appChange);
    });
  });
});
