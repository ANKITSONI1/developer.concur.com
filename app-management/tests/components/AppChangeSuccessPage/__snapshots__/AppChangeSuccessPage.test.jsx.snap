// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppChangeSuccessPage /> should render a not found page if no app or clientSecret are provided 1`] = `
<div>
  <div>
    <h2>
      Not found
    </h2>
    <Link
      className="btn bright-blue"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/"
    >
      Back to My Apps
    </Link>
  </div>
</div>
`;

exports[`<AppChangeSuccessPage /> should render the app clientId, clientSecret, and name if provided 1`] = `
<div>
  <div>
    <h2>
      Here are your updated app credentials.
    </h2>
    <p>
      <strong>
        Copy down your clientSecret. You will not be able to see it again.
      </strong>
    </p>
    <hr />
    <p>
      <strong>
        Name: 
      </strong>
      <span>
        Starbucks
      </span>
    </p>
    <p>
      <strong>
        clientId: 
      </strong>
      <code>
        test-id
      </code>
    </p>
    <p>
      <strong>
        clientSecret: 
      </strong>
      <code>
        a-client-secret
      </code>
    </p>
    <Link
      className="btn bright-blue"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/details/test-id"
    >
      Go to App Details
    </Link>
  </div>
</div>
`;
